crates/storage/engines/api.rs

```diff
@@ -216,6 +217,10 @@ pub trait StoreEngine: Debug + Send + Sync + RefUnwindSafe {
     // Used for internal store operations
     fn open_storage_trie(&self, address: Address, storage_root: H256) -> Trie;
 
-    // Get the canonical block hash for a given block number.
+    // Set the canonical block hash for a given block number.
     fn set_canonical_block(&self, number: BlockNumber, hash: BlockHash) -> Result<(), StoreError>;
+
+    fn add_local_block(&self, payload_id: u64, block: Block) -> Result<(), StoreError>;
```

Comment:
hmm, I would rename to `payload`, since block might be a bit confusing. WDYT?

Reply:
I figured block should be included in the name as the type stored is a `Block`, but I agree